core/types/celo_block.go

```diff
@@ -40,8 +44,9 @@ func (h *Header) DecodeRLP(s *rlp.Stream) error {
 
 	if preGingerbread { // Address
 		// Before gingerbread
-		decodedHeader := beforeGingerbreadHeader{}
-		err = rlp.DecodeBytes(raw, &decodedHeader)
+		decodedHeader := BeforeGingerbreadHeader{}
+		str := rlp.NewStream(bytes.NewReader(raw), uint64(len(raw)))
+		err = decodedHeader.DecodeRLP(str)
```

Comment:
The rlp package under the hood will call the `DecodeRLP` method if the type defines it, so we can simply continue to call `rlp.DecodeBytes`.

```suggestion
		err = rlp.DecodeBytes(raw, &decodedHeader)
```